traits/trait_list_object.py

```diff
@@ -797,7 +797,7 @@ def remove(self, value):
             If the value is not present.
         """
 
-        self._validate_length(len(self) - 1)
+        self._validate_length(max(len(self) - 1, 0))
```

Comment:
What about the one in `__delitem__`? 
I was looking for all occurrences of `_validate_length(X - Y)`. I see the one in `__setitem__` probably doesn't need changing.

Reply:
So the one in `__delitem__` extracts the slice first; the length of that slice is bounded by the length of the original list, so we shouldn't be able to go negative.

Reply:
Ah, but not if we have a simple index rather than a slice. Good catch. Will fix.

Reply:
Fixed! Thank you.